src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx

```diff
@@ -48,23 +46,22 @@ export const ScheduleItemNG: React.FC<ScheduleItemNGProps> = ({
   event,
   isShowFullInfo,
 }) => {
-  const { currentVenue: eventVenue } = useRelatedVenues({
+  const { currentVenue: eventVenue, relatedVenues } = useRelatedVenues({
     currentVenueId: event.venueId,
   });
-  const eventRoom = useMemo<Room | undefined>(
-    () =>
-      eventVenue?.rooms?.find((room) => {
-        const { room: eventRoom = "" } = event;
-        const noTrailSlashUrl = getUrlWithoutTrailingSlash(room.url);
-
-        const [roomName] = getLastUrlParam(noTrailSlashUrl);
-        const roomUrlParam = getUrlParamFromString(eventRoom);
-        const selectedRoom = getUrlParamFromString(room.title) === eventRoom;
-
-        return roomUrlParam.endsWith(`${roomName}`) || selectedRoom;
-      }),
-    [eventVenue, event]
-  );
+
+  const relatedVenuesRooms = relatedVenues
+    ?.flatMap((venue) => venue.rooms ?? [])
+    .filter((room) => room !== undefined);
+
+  const eventRoom = useMemo<Room | undefined>(() => {
+    const { room: eventRoomTitle = "" } = event;
+
+    return relatedVenuesRooms?.find((room) => {
+      return room.title === eventRoomTitle;
+    });
```

Comment:
Just more tidy, less boilerplate.

```suggestion
    return relatedVenuesRooms?.find(({ title }) => title === eventRoomTitle);
```

Reply:
nice, thanks. Fixed here: https://github.com/sparkletown/sparkle/pull/2183/commits/2be26c4e66c5595df4c893f818cc364b297a21b3